Webpay.Integration.CSharp/Webpay.Integration.CSharp/Response/Hosted/SveaResponse.cs

```diff
@@ -23,15 +27,53 @@ public class SveaResponse : Response
         public string ExpiryYear { get; set; }
         public string AuthCode { get; set; }
         public string Xml { get; set; }
-
+        public int MacValidation { get; set; }
+        public string Mac { get; set; }
         /// <summary>
         /// SveaResponse
         /// </summary>
         /// <param name="responseXmlBase64"></param>
         /// <exception cref="Exception"></exception>
-        public SveaResponse(string responseXmlBase64)
+        public SveaResponse(string responseXmlBase64, string mac = null, CountryCode countryCode = 0, IConfigurationProvider config = null)
+        {
+            if (responseXmlBase64 != null)
+            {
+                if (config != null)
+                {
+                    string secret = config.GetSecretWord(PaymentType.HOSTED, countryCode);
+                    if (validateMac(responseXmlBase64, secret, mac) == true)
+                    {
+                        MacValidation = 1;
+                    }
+                    else
+                    {
+                        MacValidation = 2;
+                    }
+                }
+                SetValues(responseXmlBase64);
+            }
+        }
+
+        private bool validateMac(string responseXmlBase64, string secret, string mac)
         {
-            SetValues(responseXmlBase64);
+                string macKey = stringToHashString(responseXmlBase64, secret);
```

Comment:
macKey? what does that even mean :) it is the mac calculated from our secret from the responseXmlBase64, which should correspond to the mac sent in the response -- I'd name it "calculatedMac" or something like that.